### DISCUSSION TITLE
Muncul ValueError: Cannot setitem on a Categorical with a new category, set the categories first

### DISCUSSION THREAD

**Discussion by rahmalifa** [teamalgoritma/community#28] [Q&A] [answered]
muncul error ketika menjalankan code berikut ini:
fund_raising["category"]=fund_raising["category"].fillna(value="unknown")

![image](https://user-images.githubusercontent.com/19927751/160118592-307f83b8-7906-4478-b84e-967748a859c4.png)

**Comment by hwulanayu** [comment, marked as answer]
Halo Ibu Rahma, error tersebut muncul karena Ibu menambahkan kategori baru bernama "unknown" pada kolom yang valuenya bertipe kategori yang sudah diassign sebelumnya, misal kategori sebelumnya: [minimarket, supermarket, hypermarket].
Solusi 1: Menambahkan kategori baru dengan method add_categories(). Contoh: nama_kolom.add_categories('unknown'). [Dokumentasi official fungsi add_categories](https://pandas.pydata.org/docs/reference/api/pandas.Series.cat.add_categories.html.)
Solusi 2: Melakukan imputasi sebelum mengubah tipe data ke categorical.
